submissions/franchukv/dom-api/script.js

```diff
@@ -0,0 +1,137 @@
+    },
+};
+
+function createSideNavigation() {
+    if (!mainContent.classList.contains('non-active')) {
+        footer.classList.add('footer--element-end');
+    }
+
+    startNav.classList.add('side-menu');
+
+    startHeader.classList.remove('active');
+    startNav.classList.remove('start-menu');
+    resetBtn.classList.remove('non-active');
+    burgenBtn.classList.remove('non-active');
+    mainContent.classList.remove('non-active');
+}
+
+function goToStartPage() {
+    startHeader.classList.add('active');
+    startNav.classList.add('_start--marker');
+    startNav.classList.add('start-menu');
+    resetBtn.classList.add('non-active');
+    burgenBtn.classList.add('non-active');
+    mainContent.classList.add('non-active');
+
+    startNav.classList.remove('side-menu');
+    burgenIcon.classList.remove('burger-icon--active');
+    document.body.classList.remove('_scroll-lock');
+    footer.classList.remove('footer--element-end');
+}
+
+function burgerHandler() {
+    startNav.classList.toggle('side-menu--active');
+    burgenBtn.classList.toggle('burger--white');
+    burgenIcon.classList.toggle('burger-icon--active');
+    document.body.classList.toggle('_scroll-lock');
+}
+
+function navigationHandler(event) {
+    if (startHeader.classList.contains('active') && startNav.classList.contains('start-menu')) {
+        createSideNavigation();
+        createContent(event.target.textContent);
+    }
+
+    if (startNav.classList.contains('_start--marker')) {
+        startNav.classList.remove('side-menu--active');
+        startNav.classList.remove('_start--marker');
+        burgenBtn.classList.remove('burger--white');
+        burgenIcon.classList.remove('burger-icon--active');
+    }
+
+    if (startNav.classList.contains('side-menu--active')) {
+        startNav.classList.remove('side-menu--active');
+        startNav.classList.remove('_start-marker');
+        burgenBtn.classList.remove('burger--white');
+        document.body.classList.remove('_scroll-lock');
+        burgenIcon.classList.remove('burger-icon--active');
+    }
+
+    createContent(event.target.textContent);
+}
+
+function createNavigation(item) {
+    for (item in data) {
+        let navItem = document.createElement('li');
+        navItem.classList.add('cards__item');
+
+        let h3 = document.createElement('h3');
+        h3.classList.add('cards__title');
+        h3.textContent = item;
+        navItem.appendChild(h3);
+
+        navigation.append(navItem);
```

Comment:
Adding elements to DOM from a loop is a bad practice. A browser will run reflow and repaint for every element in the loop. Instead, you can:

1. Use append method, which can add several elements in one operation.
2. Create some wrapper, add your items to the wrapper and then add it to DOM. It will be one operation.
3. Clone current container. Add items to a container and then replace your old container with a new one. But be aware of event listeners.
4. Use innerHTML instead.